src/core/public/application/types.ts

```diff
@@ -661,6 +661,28 @@ export interface InternalApplicationSetup extends Pick<ApplicationSetup, 'regist
   ): void;
 }
 
+/**
+ * Options for the {@link ApplicationStart.navigateToApp | navigateToApp API}
+ */
+export interface NavigateToAppOptions {
+  /**
+   * optional path inside application to deep link to.
+   * If undefined, will use {@link AppBase.defaultPath | the app's default path}` as default.
+   */
+  path?: string;
+  /**
+   * optional state to forward to the application
+   */
+  state?: any;
+  /**
+   * if true, will not create a new history entry when navigating (using `replace` instead of `push`)
+   *
+   * @remarks
+   * This option not be used when navigating from and/or to legacy applications.
+   */
+  replace?: boolean;
```

Comment:
@flash1293 as discussed on slack, I only implemented this option for KP apps.  (KP to KP navigation). Can you confirm that it's alright?